### PR TITLE
Allow selection of storage class provisioner in mpdev

### PR DESCRIPTION
**Pull request by vcanaa**
Introducing optional property storage_class_provisioner

Usage:
mpdev verify --deployer="$DEPLOYER" --wait_timeout="$WAIT_TIMEOUT" --image_pull_secret="$IMAGE_PULL_SECRET" --storage_class_provisioner=kubernetes.io/vsphere-volume

Supported values:
kubernetes.io/vsphere-volume
kubernetes.io/gce-pd

If no value is provided, kubernetes.io/gce-pd is used by default

### CODE REVIEW

File: scripts/verify [hunk 1]
```diff
@@ -97,14 +101,15 @@ function delete_namespace() {
...
 function clean_resources() {
-  if [[ -z "$resources_cleaned" ]]; then
-    echo "INFO cleaning all resources."
-    delete_namespace
-    if [[ ! -z "$deployer_role_name" ]]; then
-      kubectl delete clusterrolebinding "$deployer_role_name"
-    fi
-    resources_cleaned="true"
-  fi
+  # if [[ -z "$resources_cleaned" ]]; then
```
Comment (eshiroma) [inline scripts/verify:104]: Delete?

File: scripts/verify [hunk 2]
```diff
@@ -298,32 +304,32 @@ done
...
-  --timeout="$deletion_timeout" \
-  || clean_and_exit "ERROR Some resources were not deleted"
-
-echo "INFO Wait for service accounts to be deleted."
-/scripts/wait_for_deletion.sh \
-  --namespace="$NAMESPACE" \
-  --kind=serviceaccounts,roles,rolebindings \
-  --timeout="$deletion_timeout" \
-  || clean_and_exit "ERROR Some service accounts or roles were not deleted"
+# echo "INFO Stop the application"
```
Comment (eshiroma) [inline scripts/verify:307]: Replace?

File: marketplace/deployer_util/provision.py [hunk 3]
```diff
@@ -658,26 +667,32 @@ def provision_service_account(schema, prop, app_name, namespace,
...
-        'provisioner': 'kubernetes.io/gce-pd',
-        'parameters': {
-            'type': 'pd-ssd',
-        }
-    }]
-    return sc_name, add_preprovisioned_labels(manifests, prop.name)
-  else:
-    raise Exception('Do not know how to provision for property {}'.format(
-        prop.name))
+def provision_storage_class(schema, prop, app_name, namespace, provisioner):
```
Comment (eshiroma) [inline marketplace/deployer_util/provision.py:670]: Will there be an associated change to our UI provisioning flows?

Comment (vcanaa) [reply marketplace/deployer_util/provision.py:670]: Yes, the UI is trying to provision a hardcoded gce-pd as well

Comment (eshiroma) [reply marketplace/deployer_util/provision.py:670]: Makes sense. This approach seems fine for here but we probably need to make some schema changes to fully accommodate this.

**Review by eshiroma — APPROVED** [review]

File: marketplace/deployer_util/provision.py [hunk 4]
```diff
@@ -658,26 +667,32 @@ def provision_service_account(schema, prop, app_name, namespace,
...
+    parameters = {'diskformat': 'thin'}
+  elif provisioner == 'kubernetes.io/gce-pd':
+    if prop.storage_class.ssd:
+      parameters = {
+          'type': 'pd-ssd',
+      }
+    else:
+      raise Exception('Do not know how to provision for property {}'.format(
+          prop.name))
+
```
Comment (gibbleyg) [inline marketplace/deployer_util/provision.py:684]: Does there need to be an extra `else` in case the provisioner is set to another value?

**Pull request merged**